cmake/Modules/FindFLANN.cmake

```diff
@@ -120,6 +146,44 @@ find_library(FLANN_LIBRARY_DEBUG
     lib
 )
 
+find_library(FLANN_LIBRARY_DEBUG_STATIC
+  NAMES
+    flann_cpp_s-gd flann_cpp_sd
+  HINTS
+    ${PC_FLANN_LIBRARY_DIRS}
+    ${FLANN_ROOT}
+    $ENV{FLANN_ROOT}
+  PATHS
+    $ENV{PROGRAMFILES}/Flann
+    $ENV{PROGRAMW6432}/Flann
+  PATH_SUFFIXES
+    lib
+)
+
+if(FLANN_LIBRARY_SHARED AND FLANN_LIBRARY_STATIC)
```

Comment:
Is there a way to merge these 2 conditional blocks? (53-75 and 163-185)

They seem to be quite similar in nature

Reply:
The first one check TARGETS, whereas the last one use "ordinary" variables. I think its difficult to create ie. a function that can take on both. Unless I can put "targets" in a variable... But yeah, else it does the same...